### ISSUE TITLE
[Request] Add icon inside of ion-input or ion-select interface=false option 

### ISSUE THREAD

**Issue by indraraj26** [ionic-team/ionic#16653]
# Feature Request
I am having modal that will have list of country name where i can search/filter etc and on dismiss i am getting back the data and bind to ion-input text but it does not look like dropdown since ion-input doesn't have option to add arrow-down or any image inside of ion-input.

**Ionic version:**
Currently i am using ionicv3 (3.9.2) it would be great if you can add this feature in ionic4

> C:\Users\HP>ionic info
> [WARN] You are not in an Ionic project directory. Project context may be missing.
> 
> Ionic:
> 
>    ionic (Ionic CLI) : 4.4.0
> 
> System:
> 
>    NodeJS : v8.14.0
>    npm    : 6.4.1
>    OS     : Windows 10
> 
> 
> C:\Users\HP>

**Describe Preferred Solution**
you will question me why don't you bind the data to ion-select if i am closing the modal i am getting alert as default interface. or you can add action sheet, popover but there is no way to hide.

It would be great if you can implement when interface="false" nothing will popup

**Related Code**
https://stackblitz.com/edit/ionic-ftfyvp?file=pages%2Fhome%2Fhome.html

Thank you.

**Comment by paulstelzer** [comment]
Could you please explain your request in more detail? For me it sounds like this already can be done with ion-modal

**Comment by paulstelzer** [comment]
Thanks for clarify! Do you want this? -> https://material.angular.io/components/autocomplete/overview#adding-a-custom-filter

**Comment by indraraj26** [comment]
> Thanks for clarify! You want a search inside, that I did not understand from your feature request. **Do you maybe want this?** -> https://material.angular.io/components/autocomplete/overview#adding-a-custom-filter

Thank you @paulstelzer for understanding me.
Yes that i want but i cant do that in ionic do you have any alternate way ?

Thank you!

**Comment by patrickmcd** [comment]
I think I can translate, as I've wanted the same thing.

When we use `ion-select`, we have the label on the left. On the right, we have the placeholder or selected item(s), and a dropdown arrow to signify how the control will behave.

![ion-select](https://user-images.githubusercontent.com/764290/49812481-84ebb780-fd33-11e8-8bac-56a842598a66.jpg)

But if our use case is too complicated for a select, we have to create a modal that will return a value back to the parent form. Unfortunately, we don't have an input type that will match the look of the select and will seem out of place in the form.

Currently, the `ion-select` has three interface options; alert, popover, and actionsheet. @indraraj26 is asking for a 4th option of `false`. But rather than that, how about a custom function? In our template we could have `interface="mySelectHandler()"`, and in our code we would manually open a modal.

We might also need a way to add the value returned from our modal to show the selected value(s) in the select. Maybe it's more appropriate for the 4th interface option to be `modal` with the caveat that we have to manually update the select value when the modal is dismissed.

tldr; we need a way to use a modal in a form that retains the style of a select input.

**Comment by indraraj26** [comment]
yes @patrickmcd, this is what i want hope ionic team will review this and implement same in ionic v4.

Thank you for your translation and explanation!

**Comment by comfortme** [comment]
I think you can achieve same functionality with using a ion-button instead of ion-select.

**Comment by indraraj26** [comment]
***bump***
